modules/sfp_ipregistry.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+# -------------------------------------------------------------------------------
+# Name:         sfp_ipregistry
+# Purpose:      IPRegistry database query module.
+#
+# Author:      Leo Trubach <leotrubach@gmail.com>
+#
+# Created:     2020-09-08
+# Copyright:   (c) Steve Micallef
+# Licence:     GPL
+# -------------------------------------------------------------------------------
+
+import json
+import urllib.parse
+
+from spiderfoot import SpiderFootEvent, SpiderFootPlugin
+
+
+class sfp_ipregistry(SpiderFootPlugin):
+    __name__ = 'sfp_ipregistry'
```

Comment:
Why is this needed?

Reply:
If __name__ is not set the following code
```
self.sf.error(f"You enabled {self.__name__} but did not set an API key!", False)
```
outputs ugly message.

Reply:
Try this.

```python
        if self.opts["api_key"] == "":
            self.sf.error(
                f"You enabled {self.__class__.__name__} but did not set an API key!",
                False,
            )
            self.errorState = True
            return None
```

Reply:
Alternatively, hardcode the module name.

```python
        if self.opts["api_key"] == "":
            self.sf.error(
                "You enabled sfp_ipregistry but did not set an API key!",
                False,
            )
            self.errorState = True
            return None
```